src/instrument.cpp

```diff
@@ -258,39 +307,138 @@ class BytecodeRewriter {
 
     void rewrite(u8** new_class_data, int* new_class_data_len) {
         if (VM::jvmti()->Allocate(_dst_capacity, &_dst) == 0) {
-            if (rewriteClass()) {
+            const char* class_name;
+            const char* method_name;
```

Comment:
Instead of dragging output parameters through multiple functions, these can be made members of BytecodeRewriter.

Reply:
0f9931dda370582ccc37fd0c704b8e5c34e1a705